auto_gptq/nn_modules/qlinear/qlinear_cuda.py

```diff
@@ -267,10 +267,10 @@ def forward(self, x: torch.Tensor):
                     g_idx_i = self.g_idx[i*num_dim:(i+1)*num_dim]
                     weights.append(scale_i[g_idx_i.long()] * (weight_i - zeros_i[g_idx_i.long()]))
                 weights = torch.cat(weights,dim=1)
-            out = torch.matmul(x.half(), weights)
+            out = torch.matmul(x.to(weights.dtype), weights)
         out = out.half().reshape(out_shape)
```

Comment:
Why is there still a `.half()` here?

Reply:
I don't mind leaving it you if this `half()` is not a blocker for you. But was wondering given that you replaced some of the `.half()` to remove the assumption on fp16.